hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -1331,4 +1329,22 @@ private static boolean canCompare(Schema schema) {
     inflightAndCompletedPartitions.addAll(getCompletedMetadataPartitions(tableConfig));
     return inflightAndCompletedPartitions;
   }
+
+  /**
+   * Get Last commit's Metadata.
+   */
+  public static Option<HoodieCommitMetadata> getLatestCommitMetadata(HoodieTableMetaClient metaClient) {
+    try {
+      HoodieTimeline timeline = metaClient.getActiveTimeline().getCommitsTimeline().filterCompletedInstants();
+      if (timeline.lastInstant().isPresent()) {
+        HoodieInstant instant = timeline.lastInstant().get();
+        byte[] data = timeline.getInstantDetails(instant).get();
+        return Option.of(HoodieCommitMetadata.fromBytes(data, HoodieCommitMetadata.class));
+      } else {
+        return Option.empty();
+      }
+    } catch (Exception e) {
+      throw new HoodieException("Failed to get commit metadata", e);
+    }
+  }
```

Comment:
TableSchemaResolver  has a same method， why define again

Reply:
> why move this method from TableSchemaResolver to this class

@vinothchandar suggest extract this out to a separate static helper.